app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -49,17 +49,46 @@ def store_saml_request
     ).call
   end
 
+  def requested_authn_contexts
+    @requested_authn_contexts ||= saml_request.requested_authn_contexts.presence ||
+                                  [default_authn_context]
+  end
+
   def requested_authn_context
-    @requested_authn_context ||= saml_request.requested_authn_contexts.presence ||
-                                 [default_authn_context]
+    if AppConfig.env.aal_authn_context_enabled == 'true'
+      requested_aal_authn_context
+    else
+      requested_ial_authn_context
+    end
   end
 
   def default_authn_context
-    Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF
+    if AppConfig.env.aal_authn_context_enabled == 'true'
+      default_aal_context
+    else
+      default_ial_context
+    end
+  end
+
+  def default_aal_context
+    # If the service provider aal is nil, #to_i converts it to 0 which maps to default
+    Saml::Idp::Constants::AUTHN_CONTEXT_AAL_TO_CLASSREF[current_service_provider.aal.to_i]
```

Comment:
I feel like this would be clearer with an if-else? then it wouldn't need the comment

```suggestion
    if current_service_provider.aal
      Saml::Idp::Constants::AUTHN_CONTEXT_AAL_TO_CLASSREF[current_service_provider.aal.to_i]
    else
      Saml::Idp::Constants:: DEFAULT_AAL_AUTHN_CONTEXT_CLASSREF
    end
```